server/s3_get_bucket_action.cc

```diff
@@ -214,10 +216,23 @@ void S3GetBucketAction::get_next_objects_successful() {
         // Filter out by prefix
         if (kv.first.find(request_prefix) == std::string::npos) {
           // Key does not start with specified prefix; key filetered out.
-          // Check the next key.
+          // If previous/last key was matching the prefix, stop and break
+          // further enumeration.
+          if (last_key_prefix_match) {
```

Comment:
you can replace last_key_prefix_match with 
kv.first > request_prefix everywhere

Reply:
New changes compare string lexicographically